ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/CommandDeduplicationPeriodValidationIT.scala

```diff
@@ -69,6 +115,116 @@ class CommandDeduplicationPeriodValidationIT extends LedgerTestSuite {
           )
         ),
       )
-    }
+      metadataLongestDuration = extractErrorInfoMetadataValue(failure, "longest_duration")
+      // request accepted and the metadata value is valid
+      _ <- ledger.submit(
+        request.update(
+          _.commands.deduplicationDuration := DurationConversion.toProto(
+            Duration.parse(metadataLongestDuration)
+          )
+        )
+      )
+    } yield {}
+  })
+
+  test(
+    "OffsetOutsideRange",
+    "Submission with deduplication periods represented by offsets which are outside the valid range are rejected",
+    allocate(SingleParty),
+    // Canton accepts the given offsets
+    enabled =
+      _.commandDeduplicationFeatures.getDeduplicationPeriodSupport.offsetSupport.isOffsetConvertToDuration,
+    disabledReason = "Only ledgers than convert offsets to durations fail",
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    for {
+      end <- ledger.offsetBeyondLedgerEnd()
+      deduplicationPeriod = DeduplicationPeriod.DeduplicationOffset(
+        Ref.HexString.assertFromString(end.getAbsolute.toLowerCase)
+      )
+      _ <- assertFailedRequest(
+        ledger,
+        party,
+        deduplicationPeriod,
+        "Submitting a command with an invalid offset",
+        "The submitted command had an invalid deduplication period: Cannot convert deduplication offset to duration because there is no completion at given offset .*",
+        Status.Code.INVALID_ARGUMENT,
+        LedgerApiErrors.RequestValidation.InvalidDeduplicationPeriodField,
+      )
+    } yield {}
   })
+
+  test(
+    "OffsetPruned",
+    "Submission with deduplication periods represented by offsets which are pruned are rejected",
+    allocate(SingleParty),
+    // Canton accepts the given offsets
+    disabledReason = "The ledger does not support deduplication periods represented by offsets",
+    runConcurrently = false, // Pruning is involved
+    timeoutScale = 10,
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    for {
+      firstCreate <- ledger.create(party, Dummy(party))
+      firstExercise <- ledger.exercise(party, firstCreate.exerciseDummyChoice1)
```

Comment:
Why do we need exercises? Aren't creates sufficient?

Reply:
As pruning it's involved, I wanted to make sure the ACS doesn't affect this somehow.

Reply:
Ok, I trust you :)